doc/api/util.md

```diff
@@ -576,7 +576,7 @@ changes:
     codes. Colors are customizable. See [Customizing `util.inspect` colors][].
     **Default:** `false`.
   * `customInspect` {boolean} If `false`,
-    `[util.inspect.custom](depth, opts)` functions are not invoked.
+    `[util.inspect.custom](depth, opts, inspect)` functions are not invoked.
```

Comment:
Since we have the options object already part of the signature, why not make the `inspect` a new option there?

e.g.
```
object[util.inspect.custom](depth, { ...opts, inspect });
```

I'd prefer that pattern in general unless there's a specific reason you went with a new argument?

Reply:
The options are the options that the user passed through to `util.inspect()` (including all default options). Adding something that `util.inspect` does not undestand as option is therefore IMO not ideal. Example usage:

```js
util.inspect.defaultOptions.showHidden = false

const object = {
  [util.inspect.custom](depth, options, inspect) {
    assert(options.showHidden)
    const inner = { ...this, extra: true }
    return inspect(inner, options)
  }
}

util.inspect(object, { showHidden: true })
```

Reply:
Ping @jasnell